src/element/textElement.ts

```diff
@@ -261,32 +266,65 @@ const computeBoundTextPosition = (
 
 // https://github.com/grassator/canvas-text-editor/blob/master/lib/FontMetrics.js
 
-export const measureText = (text: string, font: FontString) => {
+export const measureText = (
+  text: string,
+  font: FontString,
+  lineHeight?: number,
+) => {
   text = text
     .split("\n")
     // replace empty lines with single space because leading/trailing empty
     // lines would be stripped from computation
     .map((x) => x || " ")
     .join("\n");
-
-  const height = getTextHeight(text, font);
+  const fontSize = parseInt(font);
+  const height = getTextHeight(text, fontSize, lineHeight);
   const width = getTextWidth(text, font);
 
   return { width, height };
 };
 
-const DUMMY_TEXT = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789".toLocaleUpperCase();
-const cacheApproxLineHeight: { [key: FontString]: number } = {};
-
-export const getApproxLineHeight = (font: FontString) => {
-  if (cacheApproxLineHeight[font]) {
-    return cacheApproxLineHeight[font];
+export const computeNextLineHeightForText = (
+  originalElement: NonDeleted<ExcalidrawTextElement>,
+  updatedElement: NonDeleted<ExcalidrawTextElement>,
+) => {
+  const originalLineHeight = originalElement.lineHeight;
+  // Calculate line height relative to font size
+  if (originalLineHeight === originalElement.fontSize * 1.2) {
+    return updatedElement.fontSize * 1.2;
   }
-  const fontSize = parseInt(font);
 
+  return getLegacyLineHeightForText(updatedElement);
+};
+
+// use old algorithm to calculate line height for
+// backward compatibility
+export const getLegacyLineHeightForText = (
+  textElement: ExcalidrawTextElement,
+) => {
+  const lineCount = textElement.text.replace(/\r\n?/g, "\n").split("\n").length;
```

Comment:
we have 5 of these `replace(/\r\n?/g, "\n").split("\n")` expressions. Let's factor it out into `normalizeNewlines()` (minus the split) or `splitIntoLines()` (including the split).

Reply:
Actually, we already have `noramlizeText()` https://github.com/excalidraw/excalidraw/blob/master/src/element/textElement.ts#L34-L42 which does one thing extra (`\t` -> spaces), but I think it's fine to use here. We should simplify that function's regex tho.

Reply:
I am reusing the `normalizeText` and created a new function `splitInToLines` just for splitting